### DISCUSSION TITLE
Provide for pre-authorized notifications of proximity contacts (and own notification)

### DISCUSSION THREAD

**Discussion by LeCanardQuoi** [cds-snc/covid-alert-documentation#49] [Ideas] [answered]
# Summary | Résumé
As a user of the app, I expect to be timely notified by other app users that I have been in close proximity for which the user has tested positive. However, the current percentage of users going through the extra step of “Enter your one-time key” is dismal across Canada in spite of plenty of wishful reminders (not mandated) from high officials. Arguably, diagnosed users are under the shock (if not already sick) and are far more concerned with notifying their immediate bubble and close of a kind ones, rather than attending to, mainly, strangers. They are either not doing the extra step in the app or are delaying for too long, by which time the virus has further propagated. Configuring a pre-authorization for notifying proximity contacts tracked by the app would defuse the burden on the app user under stress. The “Get a one-time key” menu entry should be enhanced for submitting to testing and automatic notification of proximity contacts should the test result return positive, in which case the extra-step would no longer be needed. The app user would therefore become directly notified of a diagnostic test results (positive, negative or else) with an appropriate message in parallel with existing test result reporting scheme. 

# Design detail | Précisions au sujet de la conception
The interface to the app server and the health care portal would have to provide for this new functionality. The app would become better tied to contact tracing, covid testing, and possibly vaccination. Maintaining the app anonymousness should remain possible. 

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet
This enhancement request does not provide for what is required on the side of the app server and health care portal. New ERs would be required against these different system components. 

The plan to increase the app adoption rate and relaunching/repurposing the app has yet to be documented, preferably accessible to the public via the github server. Further discussions are required,  preferably with the active participation of the canadian advisory council for the app. Critical leadership would be required to better integrate contact tracing, covid testing, personal privacy and possibly vaccination.

A federal, provincial or local public health authority may mandate this pre-authorization to be enabled in various “red zones” scenarios such as the collapse of traditional contact tracing, lockdowns, etc. This feature would lead to more usefulness of the app toward its notification end goal, toward higher trust into app and toward higher app adoption rates. Time being so critical when trying to limit community transmission, reducing the notification elapsed time (and removing the info flow interruption) would make a huge difference too. These "human factor" related requirements have never been properly conveyed. Critical leadership is also required on this.

**Comment by uhengart** [comment]
Your suggestion cannot be implemented within the Google/Apple framework since the framework requires explicit consent from the user (via a dialog box) before exposures are uploaded. There is nothing the app can do here to automate this process. 

An alternative suggestion is to add functionality to retrieve the test result to the app. This way, the user has more incentive to use the app and may be more likely to upload the exposures. German's app takes this approach. However, they found that only 50% of the users who would be able to upload exposures actually do so.

**Comment by LeCanardQuoi** [comment]
> Your suggestion cannot be implemented within the Google/Apple framework since the framework requires explicit consent from the user (via a dialog box) before exposures are uploaded. There is nothing the app can do here to automate this process.
> 
> An alternative suggestion is to add functionality to retrieve the test result to the app. This way, the user has more incentive to use the app and may be more likely to upload the exposures. German's app takes this approach. However, they found that only 50% of the users who would be able to upload exposures actually do so.

The preauthorized dialog is proposed as the mean to comply with the framework. Uploading exposure data remains only done if user is subsequently diagnosed positive. Yes the server or portal would now be notifying the app user of his own test result, which used to be done outside the app framework with the problems that we fully appreciate now. The proposed test result optimization would streamline the currently elusive final notifications to users having been in the proximity.

**Comment by LeCanardQuoi** [comment]
An elegant user interface scheme to implement pre-authorization would be through an activation and deactivation command of the actual alert notification. So once the user signs up to be notified upon diagnosed exposure events (whether bluetooth proximity or QR code scan), he/she also signs up to notify (preauthorize) if ever diagnosed. Fair is fair? Activation of notification inbound is currently defaulted on while outbound notification is on-demand voluntary, and too often avoided which defeats entire purpose of the app. @sidewalkballet

**Comment by obrien-j** [comment]
In keeping with our newly really release cds-snc/covid-alert-documentation#31 practices, I've moved this up into the covid-alert-documentation repo as our central point for taking input from folks outside the team, and am converting the issue into a Discussion.

**Comment by obrien-j** [comment, marked as answer]
Thanks for the suggestion @LeCanardQuoi ,

It would be great to get tighter integration with the testing centres, via each province and territory, but currently, that system is quite distributed. In the meantime, we continue to work with HC and the P/T's on how to increase the adoption of OTK's, and to reduce the burden on individuals and the public health authorities in this area.
